packages/content-slider/src/components/content-slider/content-slider.component.ts

```diff
@@ -0,0 +1,280 @@
+      const item = items[i];
+      this.setInactiveClasses(item);
+    }
+
+    this.goTo(this.scope.currentIndex);
+  }
+
+  protected onResize() {
+    throttle(() => {
+      this.debug("onResize");
+      this.getItemWidths();
+      this.update();
+    })();
+  }
+
+  protected initEventListeners() {
+    window.addEventListener("resize", this.onResize.bind(this));
+  }
+
+  protected async afterBind() {
+    await super.afterBind();
+    this.debug("afterBind", this.scope);
+
+    const contentInfoOffsetEl = this.el.querySelector<HTMLElement>(".controls");
+    if (contentInfoOffsetEl) {
+      contentInfoOffsetEl.classList.add;
+      this.addClasses(contentInfoOffsetEl, this.scope.activeColumnClasses);
+    }
+
+    this.initItems();
+    this.initEventListeners();
+  }
+
+  protected attributeChangedCallback(
+    attributeName: string,
+    oldValue: any,
+    newValue: any,
+    namespace: string | null
+  ) {
+    //E.g. transform "col-2 col-md-4" to ["col-2", "col-md-4"]
+    if (
+      attributeName === "active-column-classes" ||
+      attributeName === "inactive-column-classes"
+    ) {
+      if (typeof newValue === "string") {
+        newValue = newValue.split(" ");
+      }
+    }
+    return super.attributeChangedCallback(
+      attributeName,
+      oldValue,
+      newValue,
+      namespace
+    );
+  }
+
+  protected requiredAttributes() {
+    return [];
+  }
+
+  protected getItemElementByIndex(index: number) {
+    return this.el.querySelectorAll(".item")?.item(index);
+  }
+
+  protected addClasses(el: Element, classes: string[]) {
+    for (const curClass of classes) {
+      el.classList.add(curClass);
+    }
+  }
+  protected removeClasses(el: Element, classes: string[]) {
+    for (const curClass of classes) {
+      el.classList.remove(curClass);
+    }
+  }
+
+  protected setInactiveClasses(el: Element) {
+    this.removeClasses(el, [
+      ...this.scope.activeColumnClasses,
+      this.scope.activeClass,
+    ]);
+
+    this.addClasses(el, this.scope.inactiveColumnClasses);
+  }
+
+  protected setActiveClasses(el: Element) {
+    this.removeClasses(el, this.scope.inactiveColumnClasses);
+
+    this.addClasses(el, [
+      ...this.scope.activeColumnClasses,
+      this.scope.activeClass,
+    ]);
+  }
+
+  public next() {
+    const newActiveIndex = this.scope.currentIndex + 1;
+    this.goTo(newActiveIndex);
+  }
+
+  public previous() {
+    const newActiveIndex = this.scope.currentIndex - 1;
+    this.goTo(newActiveIndex);
+  }
+
+  public goTo(newActiveIndex: number) {
+    if (!this.el && this.el !== null) {
+      console.error("Component is not ready!");
+      return;
+    }
+
+    this.debug("goTo");
+
+    this.getItemWidths();
+
+    const oldActiveItem = this.el.querySelector(".active");
+    const newActiveItem = this.getItemElementByIndex(newActiveIndex);
+    if (oldActiveItem) {
+      this.setInactiveClasses(oldActiveItem);
+    } else {
+      console.warn("No old active item found!");
```

Comment:
Diese Warnung tritt momentan immer ganz zu Anfang auf, weil noch kein Item die `active`-Klasse erhalten hat.
`setActiveItem(0)` wurde ausgeführt, aber das setzt ja nicht die Klasse, sondern das wird separat in `setActiveClasses` getan. Wäre vielleicht gut, diese Funktionen zusammenzuführen?

Reply:
Habe ich nun behoben indem diese Klasse innerhalb der Methode `initItems` gesetzt wird